src/SfxWeb/src/app/modules/backup-restore/storage-form/storage-form.component.ts

```diff
@@ -71,10 +84,43 @@ export class StorageFormComponent implements OnInit {
 
       storage.get('Path').setValidators([Validators.required]);
     }
-
+    this.updateStorageKindValidatorsPrimaryCredentials(storage, false);
+    this.updateStorageKindValidatorsSecondaryCredentials(storage, false);
     storage.get('ContainerName').updateValueAndValidity();
     storage.get('ConnectionString').updateValueAndValidity();
     storage.get('Path').updateValueAndValidity();
   }
+  updateStorageKindValidatorsPrimaryCredentials(storage: AbstractControl, IsEmptyPrimaryCredential: boolean){
+    if(IsEmptyPrimaryCredential === true)
+    {
+      storage.get('PrimaryUserName').setValidators(null);
+      storage.get('PrimaryPassword').setValidators(null);
+      storage.get('PrimaryUserName').setValue('');
+      storage.get('PrimaryPassword').setValue('');
+    }
+    else
+    {
+      storage.get('PrimaryUserName').setValidators([Validators.required]);
+      storage.get('PrimaryPassword').setValidators([Validators.required]);
+    }
+    storage.get('PrimaryUserName').updateValueAndValidity();
+    storage.get('PrimaryPassword').updateValueAndValidity();
+  }
 
+  updateStorageKindValidatorsSecondaryCredentials(storage: AbstractControl, IsEmptySecondaryCredential: boolean){
+    if(IsEmptySecondaryCredential === true)
```

Comment:
remove === true to simplify this.

Reply:
Thanks. I have made the necessary changes